src/Image/Darkroom/ImageMagick.php

```diff
@@ -72,8 +72,47 @@ protected function convert(string $file, array $options): string
 			}
 		}
 
+		// frame option to allow selecting layers for multi-layer or frames for animated images
+		$fileOptions = '';
+		$frame = $options['frame'];
+		$maxFrames = $this->frameCount($file, $options);
+		if ($frame !== null) {
+			// check whether frame is in bounds
+			if ($frame < 0) {
+				throw new Exception('Frame option must be a positive integer');
+			}
+
+			if ($frame !== 0 && $frame >= $maxFrames) {
+				throw new Exception('Frame option must be smaller than the number of frames in the image');
+			}
```

Comment:
Contemplating whether it would be smoother to just clop the bounds to 0/max instead of throwing exceptions.

Reply:
I decided to throw an exception here because I remembered #4249 and thought it was closed already.

Reply:
But with throwing an exception we would even put more fire to #4249 because then Kirby would simply deliver the source image. I feel delivering the first/last frame when requesting a frame out of bounds might still be better than delivering the source image?